CRM/Activity/BAO/Activity.php

```diff
@@ -2140,6 +2140,10 @@ public static function getActivitySubject($entityObj) {
         $membershipType = CRM_Member_PseudoConstant::membershipType($entityObj->membership_type_id);
         $subject = $membershipType ? $membershipType : ts('Membership');
```

Comment:
If `$membershipType` is an array at this point casting it to string would just produce the string "Array". 

The cause of this would seem to be a messy implementation of `CRM_Member_PseudoConstan::tmembershipType` which can return an array, NULL or a string, as well as calling a deprecated function.

It might be outside the scope of this but fixing that method, possibly splitting it into one that fetches all types, and one that fetches a single for an ID, would seem like a nicer solution.